tests/errors/pure_warnings.py

```diff
@@ -0,0 +1,22 @@
+# mode: error
+# tag: warnings
+
+import cython
+import typing
+from cython.cimports.libc import stdint
+
+
+def main():
+    foo1: typing.Tuple = None
+    foo1: typing.Bar = None
+    foo2: Bar = 1  # warning
+    foo3: int = 1
+    foo4: cython.int = 1
+    foo5: stdint.bar = 5  # warning
+    foo6: object = 1
```

Comment:
Here, I wanted also to test the following case:
```python
foo7: cython.bar = 1
```
But it yields following warnings:
```
14:16: Unknown type declaration 'cython.bar' in annotation, ignoring
25:10: 'cpdef_method' redeclared
36:10: 'cpdef_cname_method' redeclared
977:29: Ambiguous exception value, same as default return value: 0
1018:46: Ambiguous exception value, same as default return value: 0
1108:29: Ambiguous exception value, same as default return value: 0
```

Not sure why there are additional warnings. It is the reason why it is not included in the test...

Reply:
That looks like utility code, but I'm not sure from what.

(Will have a proper look at this PR at some point fairly soon...)

Reply:
Yes, sadly, we have a couple of utility code files that report warnings. We should eventually make those internal (or, rather, fix them).

Reply:
So I think what's happening here is that missing an attribute lookup on `cython` causes it to cause `CythonScope.load_cythonscope` which loads "TestClass" and some memoryview code. I think this is harmless (because it does get dropped) but does cause the warnings. Running it outside the test suite gives:

```
warning: TestClass:25:10: 'cpdef_method' redeclared 
warning: TestClass:36:10: 'cpdef_cname_method' redeclared 
warning: View.MemoryView:977:29: Ambiguous exception value, same as default return value: 0
warning: View.MemoryView:1018:46: Ambiguous exception value, same as default return value: 0
warning: View.MemoryView:1108:29: Ambiguous exception value, same as default return value: 0
```

What we've done before: in the `_WARNINGS` string you can have comments and the warnings don't need to be in order. So just add these warnings with a comment at the end of the list of warnings:

```
# Spurious warnings from utility code - not part of the core test
25:10: 'cpdef_method' redeclared
36:10: 'cpdef_cname_method' redeclared  
977:29: Ambiguous exception value, same as default return value: 0
1018:46: Ambiguous exception value, same as default return value: 0
1108:29: Ambiguous exception value, same as default return value: 0
"""
```

Reply:
Fixed as proposed by @da-woods.